### ISSUE TITLE
Calling hide then show with multiple modals

### ISSUE THREAD

**Issue by Steiny69** [uikit/uikit#2123]
**TLDR;** Calling hide and then show when there are two modas hides them both. Calling show needs to remove any registered events or the event listener needs additional logic to prevent this.

I am using UIkit with React JS and found it a little tricky to use a component's state to show or hide a modal dialog. In case you're not familiar with React JS the idea here is that there is a JS object which maintains it's own state that is then accessed in a render function. The render function would then generate the DOM elements to be displayed. For example:
```
render() {
  let styles = this.state.isOpen ? " uk-open" : "";
  return <div className={"uk-modal" + styles}>
     ...
  </div>;
}
```
Of course this doesn't work since just setting the style is not sufficient. The workaround was to hook into other parts of the lifecycle although it is still a little ugly (and possibly has race conditions - TBC). It ends up looking something like:
```
  componentDidMount() {
    this.displayModal();
  }

  componentDidUpdate() {
    this.displayModal();
  }

  displayModal() {
    let modal = UIkit.modal("#my-modal");
    if (this.props.open) {
      modal.show();
    } else {
      modal.hide();
    }
  }
```
This works ok until there is another modal on the page. What happens then is that `hide` is called on this modal, call it modalA, which registers the transition end event listener. When the state of this component changes to be open it will call `show` which of course hides the current modal, call it modalB, and then shows modalA. However once modalB finishes its transition it triggers both callbacks and both modalA and modalB are hidden.

**Comment by philoest** [comment]
We have released the next major version, UIkit3. To start focusing our work on the most recent release, we decided to close all the issues related to UIkit2. Please check, if these issues still persist in version 3 and if so, please open a new issue. 

Thanks for your understanding,
Your UIkit Team
